### PR TITLE
User manual documentation

### PR DESCRIPTION
**Pull request by jparisu**
Done in this PR:

- [x] Add initial page picture
- [x] User manual commands
- [x] User manual format
- [x] User manual interactive CLI
- [x] User manual one-shot
- [x] User manual YAML configuration

### CODE REVIEW

File: docs/rst/user_manual/commands/reader.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+Arguments
+=========
+
+**Readers** command support 0 or 1 argument:
+
+*No argument*
+-------------
+
+When no arguments are given to this command, the information shown is a **list** with every DataReader currently active in the network, giving their Guid,
```
Comment (irenebm) [inline docs/rst/user_manual/commands/reader.rst:26]:
```suggestion
When no arguments are given to this command, the information shown is a **list** with every DataReader currently active in the network, giving their :term:`Guid <Guid>`,
```

File: docs/rst/user_manual/commands/reader.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+the name of their respective DomainParticipants, and the topic name and topic data type name.
```
Comment (irenebm) [inline docs/rst/user_manual/commands/reader.rst:27]:
```suggestion
the name of their respective :term:`DomainParticipants <DomainParticipants>`, and the topic name and topic data type name.
```

File: docs/rst/user_manual/commands/data.rst [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+Key-words
+=========
+
+These are the key-words recognize as this command:
+``show`` ``print`` ``s`` ``S``.
+
+Data Type discovered
+====================
+
+In order for a :term:`Topic` to be printable by the application, the |spy| requires to know the Data Type of such Topic.
```
Comment (irenebm) [inline docs/rst/user_manual/commands/data.rst:27]:
```suggestion
In order for a :term:`Topic` to be printable by the application, the |spy| requires to know the Data Type of such topic.
```

File: docs/rst/user_manual/commands/data.rst [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+The information whether the topic data type is already discovered be the application can be shown by using :ref:`user_manual_command_topic` command.
```
Comment (irenebm) [inline docs/rst/user_manual/commands/data.rst:28]:
```suggestion
The information whether the Topic Data Type is already discovered be the application can be shown by using :ref:`user_manual_command_topic` command.
```

File: docs/rst/user_manual/configuration.rst [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+    |spy| entities are created with the QoS of the first Subscriber found in this Topic.
+
+|spy| allows filtering of DDS :term:`Topics<Topic>`, that is, it allows to define which DDS Topics are going to be
+relayed by the application.
+This way, it is possible to define a set of rules in |spy| to filter those data samples the user does not wish to
+forward.
+
+It is not mandatory to define such set of rules in the configuration file. In this case, a |spy| will read all
+the data published under the topics that it automatically discovers within the DDS network to which it connects.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/configuration.rst:58]: The format is to link with glossary only first appearance of each word. Thus, I add :term: in line 34.

**Review by irenebm — APPROVED** [review]
LGTM

**Pull request merged**